From: Chengguang Xu <cgxu519@gmx.com>
Date: Tue, 13 Mar 2018 10:43:45 +0800
Subject: ceph: return proper bool type to caller instead of pointer
Git-commit: 98cfda81040f4a6925a98567d036a918e855d4ec
Patch-mainline: v4.17-rc1
References: bsc#1089115

Change to return true/false only for bool type return code.

Signed-off-by: Chengguang Xu <cgxu519@gmx.com>
Reviewed-by: "Yan, Zheng" <zyan@redhat.com>
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/mds_client.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/fs/ceph/mds_client.c
+++ b/fs/ceph/mds_client.c
@@ -418,9 +418,10 @@ struct ceph_mds_session *__ceph_lookup_m
 
 static bool __have_session(struct ceph_mds_client *mdsc, int mds)
 {
-	if (mds >= mdsc->max_sessions)
+	if (mds >= mdsc->max_sessions || !mdsc->sessions[mds])
 		return false;
-	return mdsc->sessions[mds];
+	else
+		return true;
 }
 
 static int __verify_registered_session(struct ceph_mds_client *mdsc,
